warehouse/ingest-core/src/main/java/datawave/ingest/data/normalizer/SimpleGroupFieldNameParser.java

```diff
@@ -35,20 +31,41 @@ public SimpleGroupFieldNameParser(boolean shouldIncludeDelimInGroupName) {
      */
     public NormalizedContentInterface extractFieldNameComponents(NormalizedContentInterface origField) {
         String originalIndexedFieldName = origField.getIndexedFieldName();
-        String normParts[] = StringUtils.split(originalIndexedFieldName, '.');
-        int index = originalIndexedFieldName.indexOf('.');
+        String[] splits = StringUtils.split(originalIndexedFieldName, '.');
+        trimmed = false;
         
-        if (index > -1) {
-            String baseFieldName = originalIndexedFieldName.substring(0, index);
-            String group = originalIndexedFieldName.substring(index + (shouldIncludeDelimInGroupName ? 0 : 1));
+        if (splits.length > 1) {
+            // we are nested, i.e. we have a field parent names in the event field name
             
             NormalizedFieldAndValue revisedField = new NormalizedFieldAndValue(origField);
             
-            // we are nested
+            String baseFieldName = null;
+            String origGroup = null;
+            String group = null;
+            String subgroup = null;
+            
+            baseFieldName = splits[0];
+            origGroup = extractGroup(splits);
+            
+            group = trimGroup(origGroup);
+            
+            if (!trimmed) {
```

Comment:
won't trimmed always be false here?

Reply:
I am guessing that trimmed should be set to (!group.equals(origGroup)) ?

Reply:
Oh, I see what you are doing.  I do not like the parser holding state however.  I think you can remove that variable and just test for group.equals(origGroup).

Reply:
No, I tried that first, thanks.  It's not true that the trimmed value will equal the original for this purpose

Reply:
Then we need to return the trimmed boolean back with the method as well as the resulting group.
Perhaps pass in a StringBuilder for the trimmed result and the method should return a boolean as to whether it was "trimmed".